### DISCUSSION TITLE
How webpack-assets-manifest plugin defines names of assets?

### DISCUSSION THREAD

**Discussion by krutoo** [webdeveric/webpack-assets-manifest#267] [Q&A] [answered]
Hi, in my project i use webpack with module federation plugin and webpack-assets-manifest plugin

webpack-assets-manifest plugin works properly but i dont undesrtand how it defines "original" names of assets

For example i use Module Federation plugin with defined option `filename` like this:
```js
new webpack.container.ModuleFederationPlugin({
  name: "foobar",
  filename: "remote-entry.js",
  exposes: {
    "./Header": "./src/components/header/index.tsx",
  },
}),
```

but in assets manifest i see this:
```json5
{
  "foobar.js": "remote-entry.js",
  "remote-entry.js.map": "remote-entry.js.map",
  // ...some other assets
}
```

Keys of manifest object for remote entry relative assets are little bit inconsistent

How webpack-assets-manifest plugin or maybe webpack itself defines names of assets?

**Comment by webdeveric** [comment]
@krutoo I've not used `ModuleFederationPlugin` before so I don't know how or why there would be any inconsistencies. What do you expect to be in the assets manifest for your configuration?

Can you create minimal reproducible example that I can clone so I can try some things?

**Reply by krutoo** [reply]
I think that key `foobar.js` must be `remote-entry.js` like in case of .map file but im not sure

You can just clone this repo https://webpack.js.org/plugins/mini-css-extract-plugin/#extracting-all-css-in-a-single-file

and:

- remove optimization property of config
- add webpack-assets-manifest plugin

**Reply by webdeveric** [reply]
@krutoo The `.js` suffix is being added to the `chunkName` here: https://github.com/webdeveric/webpack-assets-manifest/blob/dde2bc3bd342bfffd8318d6f2d0c030cdb21599e/src/WebpackAssetsManifest.js#L351
The `chunkName` for your use case is the `name` option passed into the `ModuleFederationPlugin`.

**Reply by krutoo** [reply]
@webdeveric  thanks, but why didn't chunkName apply for the .map?

**Reply by webdeveric** [reply, marked as answer]
@krutoo I get data from `compilation.getAssets()` and use that to determine what goes in the assets manifest. Some assets do not have `info.sourceFilename` since they are generated files. In that case, I use the `name` reported from `getAssets()`.

https://github.com/webdeveric/webpack-assets-manifest/blob/dde2bc3bd342bfffd8318d6f2d0c030cdb21599e/src/WebpackAssetsManifest.js#L524

https://github.com/webdeveric/webpack-assets-manifest/blob/dde2bc3bd342bfffd8318d6f2d0c030cdb21599e/src/WebpackAssetsManifest.js#L574-L576
